addons/account/models/account.py

```diff
@@ -334,6 +334,19 @@ def unlink(self):
         partner_prop_acc = self.env['ir.property'].search([('value_reference', 'in', values)], limit=1)
         if partner_prop_acc:
             raise UserError(_('You cannot remove/deactivate an account which is set on a customer or vendor.'))
+        default_journals = self.env['account.journal'].search([
+            ('default_credit_account_id', 'in', self.ids),
+            ('default_debit_account_id', '=', self.ids)
+        ])
+        if default_journals:
+            used_accounts = default_journals.mapped('default_credit_account_id')
+            used_accounts |= default_journals.mapped('default_debit_account_id')
+            used_accounts = used_accounts & self
+            raise UserError(
+                _("The account %r is already used in journal %r and cannot be removed.")
```

Comment:
need a better error message, if you have more than one case it will be difficult to read:
`The account Product Sales, Expenses is already used in journal Customer Sales, Vendor Bills and cannot be removed` is not really clear